comp/core/internal/params.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package internal
+
+import (
+	"github.com/djmitche/dd-agent-comp-experiments/pkg/util/startup"
+)
+
+// BundleParams must be defined here to avoid package dependency cycles.
+
+// BundleParams defines the parameters for this bundle.
+type BundleParams struct {
+	// ConfFilePath is the path to the configuration file.
+	ConfFilePath string
+
+	// AutoStart determines whether components in this bundle should start
+	// automatically.  This is typically true for long-running processes and
+	// false for one-shot processes.  This defaults to Always.
+	AutoStart startup.AutoStart
+
+	// Console determines whether log messages should be output to the console.
+	Console bool
+}
```

Comment:
Let's see how that works out.  I hope it's self-limiting!

Reply:
The number is small enough now but we should keep an eye how this evolve. `5` parameters could be a maximum value (arbitrary value). Beyond a certain value, it is probably better that each app defines what it needs as a list of components or bundles  instead of relying on flags.